Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -3126,14 +3136,14 @@ private void OnIbAutomaterOutputDataReceived(object sender, OutputDataReceivedEv
             // login failed
             if (e.Data.Contains("Login failed"))
             {
-                _loginFailed = true;
+                _stateManager.LoginFailed = true;
                 _ibAutomaterInitializeEvent.Set();
             }
 
-            // an existing session was detected and IBAutomater clicked the "Exit Application" button
-            else if (e.Data.Contains("Existing session detected") && !_performingRelogin)
+            // an existing session was detected
+            else if (e.Data.Contains("Existing session detected"))
             {
-                _existingSessionDetected = true;
+                _stateManager.ExistingSessionDetected = true;
```

Comment:
It seems `LoginFailed`, `ExistingSessionDetected` and `SecurityDialogDetected` all trigger termination. What do you think if we had a single error code flag? Which we set with defined enum values, with the objective of simplifying the state machine